Source/BSN.Commons/Infrastructure/MessageBroker/EventAggregator/InMemoryEventAggregatorSubscriptionManager.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.Contracts;
+using IEvent = BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.Contracts.IEvent;
+
+namespace BSN.Commons.Infrastructure.MessageBroker.EventAggregator
+{
+    public partial class InMemoryEventAggregatorSubscriptionManager : IEventAggregatorSubscriptionManager
+    {
+        public InMemoryEventAggregatorSubscriptionManager()
+        {
+            _eventReceivers = new Dictionary<string, IEventReceiver>();
+            _eventsType = new List<Type>();
+        }
+
+        public void AddSubscription<TEvent>(IEventReceiver eventReceiver) where TEvent : IEvent
```

Comment:
Please add comments for all public method and public classes :)

Reply:
Please follow https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/examples

Reply:
done.